### ISSUE TITLE
Limitation of current log sub-system

### ISSUE THREAD

**Issue by rit001** [traefik/traefik#10299]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you expect to see?

Background

The Common Log Format (CLF) description used by Traefik is currently hardcoded to an exact format with exact fields selected, with only the JSON output having some control over what will or will not be included in the resulting output. This is problematic as just about all reporting tools are coded to use CLF as are intrusion detection tools, so the JSON option has limited usability.

The problem

Once a deployment of Traefik is fronted by a CND service such as Cloudflare the IP address placed into the log file is that of the Cloudflare server, not the originator and there is currently no way to change this.

Requested solution

Could a flag be added that causes the logged remote_IP_address to be set to the value of the X-Forwarded-For header field if defined? 

Security concerns 

It would make sense to only active such a setting if both the following have been set

                 entrypoints.http.forwardedHeaders.trustedIPs 
                 entrypoints.https.forwardedHeaders.trustedIPs

Having these set are a good indication that an upstream system such as a CDN is involved. Without such a system in place, it would be possible for a connecting system to spoof IP address values by setting X-Forwarded-For to any value they wish.

Other solutions I have considered (and discounted)

- Place a more flexible reverse proxy between Traefik and the Application so that I can generate valid logs. This is not a valid solution as I have 5 applications to frontend and the configuration becomes something of a mess when having Traefik call a single instance of Apache or Nginx that frontends many applications so that a single log file can be generated.

- Pass the Traefik JSON log in real-time so that I can output a CLF-based version. The overhead of doing this is just too great as the JSON format requires far more I/O to create and processing power during its generation and processing.

**Comment by unknowndevQwQ** [comment]
Hi, I've run into a similar problem, I'd like to see some http headers of connections in the traefik logs, but the clf is always fixed (even after adding the appropriate configurations) and the format of the traefik's clf logs is not as intuitive as caddy's console format.

**Comment by lbenguigui** [comment]
Hello @rit001,

Thanks for your interest in Traefik !

When a request come from a trusted IP, the X-Forwarded-For header is kept and eventually used in the remote_IP_address  field. So unless we missed something, Traefik already behaves as you want.

I will close this issue, don't hesitate to re-open it if needed.

**Comment by rit001** [comment]
OK, that is not something I noticed when working with the logs, and I did not see it noted in the docs (hence my post). So it may be best to make the docs clearer that this happens automatically.

This is all going to get a little more complicated shortly as Google currently seems to be proposing to deploy automatic proxies for Chrome users. If they do this it will then break the X-Forwarded-For header anyway as all a vendor such as Cloudflare will see is the IP address of the proxy.

Thanks
